homeassistant/components/enigma2/media_player.py

```diff
@@ -0,0 +1,251 @@
+    vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
+    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_PREFER_PICON, default=DEFAULT_PREFER_PICON): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up of an enigma2 media player."""
+    if discovery_info:
+        discovered_config = {
+            CONF_NAME: DEFAULT_NAME,
+            # Do not use the discovered port, as it is the streaming
+            # service port (8001) which is not what we want.
+            CONF_PORT: DEFAULT_PORT,
+            CONF_HOST: discovery_info['host'],
+            CONF_USERNAME: DEFAULT_USERNAME,
+            CONF_PASSWORD: DEFAULT_PASSWORD,
+            CONF_SSL: DEFAULT_SSL,
+            CONF_PREFER_PICON: DEFAULT_PREFER_PICON,
+        }
+        add_devices([Enigma2Device(discovery_info['hostname'],
+                                   discovered_config)], True)
+        return
+
+    add_devices([Enigma2Device(config[CONF_NAME], config)], True)
+
+
+class Enigma2Device(MediaPlayerDevice):
+    """Representation of an Enigma2 box."""
+
+    def __init__(self, name, config):
+        """Initialize the Enigma2 device."""
+        import openwebif.api
+        self._state = None
+        self._name = name
+        self.e2_box = \
+            openwebif.api.CreateDevice(host=config[CONF_HOST],
+                                       port=config[CONF_PORT],
+                                       username=config[CONF_USERNAME],
+                                       password=config[CONF_PASSWORD],
+                                       is_https=config[CONF_SSL],
+                                       prefer_picon=config[CONF_PREFER_PICON])
+
+        self.volume = 10
+        self.current_service_channel_name = None
+        self.current_programme_name = None
+        self.current_service_ref = None
+        self.muted = False
+        self.picon_url = None
+        self.status_info = {}
+        self.sources = self.e2_box.sources
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def supported_features(self):
+        """Flag of media commands that are supported."""
+        return SUPPORTED_ENIGMA2
+
+    def turn_off(self):
+        """Turn off media player."""
+        self.e2_box.toggle_standby()
+
+    def turn_on(self):
+        """Turn the media player on."""
+        self.e2_box.toggle_standby()
+
+    @property
+    def media_title(self):
+        """Title of current playing media."""
+        return self.current_service_channel_name
+
+    @property
+    def media_series_title(self):
+        """Return the title of current episode of TV show."""
+        if self.current_service_ref.startswith('1:0:0'):
```

Comment:
After seeing your comment about adding things into the pypi library, a nicer way to handle this would be to modify the library to provide `is_live` or `is_recording` (if that's what the red dot means).

Reply:
Done now in the pypi module.